### ISSUE TITLE
FEAT: Improve of Filtering.

### ISSUE THREAD

**Issue by repiatx** [beekeeper-studio/beekeeper-studio#2129]
Hi Guys,

I'm always frustrated when i try to search something on table data. Like i want to search name "nizamettin" with "niz" or "zam". Can you add something for this because "Apply Filter" doesnt allow me search something like this.

If you could make this feature i'll buy it.

**Comment by azmy60** [comment]
Hey @repiatx ,

If you open a table, you can search for data using `like` condition and use `%` like this:

![image](https://github.com/beekeeper-studio/beekeeper-studio/assets/38707148/191fd5ef-e56e-4bc0-90ce-fe9193518970)

Is that what you're looking for?

**Comment by repiatx** [comment]
> Hey @repiatx ,
> 
> If you open a table, you can search for data using `like` condition and use `%` like this:
> 
> ![image](https://private-user-images.githubusercontent.com/38707148/322740437-191fd5ef-e56e-4bc0-90ce-fe9193518970.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMyNjU5NTIsIm5iZiI6MTcxMzI2NTY1MiwicGF0aCI6Ii8zODcwNzE0OC8zMjI3NDA0MzctMTkxZmQ1ZWYtZTU2ZS00YmMwLTkwY2UtZmU5MTkzNTE4OTcwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDE2VDExMDczMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZmZGJjMzAzN2I0NWM4NDNjZTJiOTkwNzQ4MTZhZjM1Y2M4M2RkYzQ5YTAyYjAxZjFmYjdiNzY1N2JiZDIyMDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.z534wXKK1xfJ9NaOa69qrNv-GZib8dE2JtcONbW7aO4)
> 
> So you can do something like `niz%` for starts with niz, or `%zam%` for contains zam. Is that what you're looking for?

i know that but even without querying i want to accomplish that. Because there is data here. I want to search on opened datas.

**Comment by rathboma** [comment]
Oh, you want to find something within the visible table?

**Comment by rathboma** [comment]
For implementation - this could work similarly to how `find in sheet` works in google sheets. Hit ctrl/cmd+f in google sheets and it pops open a 'find in sheet' box. We could do the same thing for the visible view. This might even be supported by tabulator?

It should: have a placeholder 'find in visible data', and up/down arrows
When searching it should highlight the cells with the value in them
The highlighting should be a different color vs the `add new row` color


![image](https://github.com/beekeeper-studio/beekeeper-studio/assets/279769/1552e65a-e70b-42b8-ab10-1ebde73147c6)


![image](https://github.com/beekeeper-studio/beekeeper-studio/assets/279769/c8b8cbe9-d84b-4dbe-b758-54436eded719)

**Comment by azmy60** [comment]
I can't find that in tabulator, except the `.setFilter`, but that will filter rows that don't match. I think we'll need a custom module for this.

**Comment by repiatx** [comment]
> For implementation - this could work similarly to how `find in sheet` works in google sheets. Hit ctrl/cmd+f in google sheets and it pops open a 'find in sheet' box. We could do the same thing for the visible view. This might even be supported by tabulator?
> 
> It should: have a placeholder 'find in visible data', and up/down arrows When searching it should highlight the cells with the value in them The highlighting should be a different color vs the `add new row` color
> 
> ![image](https://private-user-images.githubusercontent.com/279769/322887861-1552e65a-e70b-42b8-ab10-1ebde73147c6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMzMzA0NDEsIm5iZiI6MTcxMzMzMDE0MSwicGF0aCI6Ii8yNzk3NjkvMzIyODg3ODYxLTE1NTJlNjVhLWU3MGItNDJiOC1hYjEwLTFlYmRlNzMxNDdjNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQxN1QwNTAyMjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xMTM5NWNlYWNkYzE1ODhlOGU1NDZmYmQyYzNhMTE0ZmJlMTYzOTMxMTRmMWIwNjYzMTg4YTkwODZlYzlmNzdmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.A_4bK1gdmGLhRIIKf1IOOZLS28J69gC20N8k-n9ZfCc)
> 
> ![image](https://private-user-images.githubusercontent.com/279769/322887789-c8b8cbe9-d84b-4dbe-b758-54436eded719.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMzMzA0NDEsIm5iZiI6MTcxMzMzMDE0MSwicGF0aCI6Ii8yNzk3NjkvMzIyODg3Nzg5LWM4YjhjYmU5LWQ4NGItNGRiZS1iNzU4LTU0NDM2ZWRlZDcxOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQxN1QwNTAyMjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lNjZhOTljN2U3ODMwYWZkMmQ2OWE5MmViNDJmMzc2YmM4ODk5NWIxYmNkNWIwYWZhOTgwZWJkOWZmYjhiYjU3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.hlDcYGV3KhtOeqzeCbQKFI28TSFBWgpuYAVXREj28HU)

Yes actually but please dont just hightlight it. go to that line too please :)

**Comment by icon2341** [comment]
Hi, this feature is really important to me and basically the only reason I use datagrip by Jetbrains

**Comment by tomh4** [comment]
same here, i am so happy with beekeeper but the fact that there is no find and replace makes me switch back to dbeaver again...
